pkg/apis/core/types_seed.go

```diff
@@ -274,6 +274,15 @@ type SeedSettingVerticalPodAutoscaler struct {
 	// is enabled by default because Gardener heavily relies on a VPA being deployed. You should only disable this if
 	// your seed cluster already has another, manually/custom managed VPA deployment.
 	Enabled bool
+
+	// MaxAllowed specifies the maximum amount of resources that will be recommended for Gardener components. The value is applied
+	// by setting it in the MaxAllowed field of the component's VPA object. If the VPA has a default value
+	// for MaxAllowed, the smaller of the two values is used. If this field is nil, no change is made to the MaxAllowed
+	// values in component VPAs.
+	//
+	// The effects of the setting are currently limited to the following shoot control plane components:
+	//  etcd, kube-apiserver,prometheus
+	MaxAllowed corev1.ResourceList
```

Comment:
```suggestion
	MaxAllowed *corev1.ResourceList
```

Reply:
corev1.ResourceList is a type alias to `map[ResourceName]resource.Quantity`. Ref https://github.com/kubernetes/kubernetes/blob/0a406ee1b4b65574c02f8aec6ff5a69d07b68be6/staging/src/k8s.io/api/core/v1/types.go#L6340

Hence, we don't need to use a pointer. It is already a reference type.